test/EVMHost.cpp

```diff
@@ -0,0 +1,253 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * EVM execution host, i.e. component that implements a simulated Ethereum blockchain
+ * for testing purposes.
+ */
+
+#include <test/EVMHost.h>
+
+#include <test/evmc/helpers.hpp>
+#include <test/evmc/loader.h>
+
+#include <libdevcore/Exceptions.h>
+#include <libdevcore/Assertions.h>
+#include <libdevcore/Keccak256.h>
+#include <libdevcore/picosha2.h>
+
+using namespace std;
+using namespace dev;
+using namespace dev::test;
+
+namespace
+{
+
+
+evmc::vm& getVM()
+{
+	static unique_ptr<evmc::vm> theVM;
+	if (!theVM)
+	{
+		// TODO make this an option
+		for (auto path: {
+			"deps/lib/libevmone.so",
+			"../deps/lib/libevmone.so",
+			"/usr/lib/libevmone.so",
+			"/usr/local/lib/libevmone.so",
+			// TODO the circleci docker image somehow only has the .a file
+			"/usr/lib/libevmone.a"
+		})
```

Comment:
Also please add a `libevmone.dylib` version for macos.

Reply:
This is just for fuzzing - we'll have to find a much nicer version for general tests anyway.